apps/files_sharing/lib/Controller/ShareesAPIController.php

```diff
@@ -152,6 +166,16 @@ public function __construct(
 	 */
 	public function search(string $search = '', string $itemType = null, int $page = 1, int $perPage = 200, $shareType = null, bool $lookup = false): DataResponse {
 
+		// if some groups are excluded, check the user is allowed to share
+		if ($this->config->getAppValue('core', 'shareapi_exclude_groups', 'no') === 'yes') {
+			$excludedGroups = (array)json_decode($this->config->getAppValue('core', 'shareapi_exclude_groups_list', ''), true);
+			$usersGroups = $this->groupManager->getUserGroupIds($this->userManager->get($this->userId));
+			if (array_intersect($usersGroups, $excludedGroups) === $usersGroups) {
+				return new DataResponse($this->result);
+			}
```

Comment:
1. If `$excludedGroups = ['A']` and `$usersGroups = ['A']`, then this returns early. OK
2. If `$excludedGroups = ['A']` and `$usersGroups = ['B']`, then this does not return early. OK
3. If `$excludedGroups = ['A']` and `$usersGroups = ['A', 'B']`, then this does not return early. NOT OK ?
4. If `$excludedGroups = ['A', 'B']` and `$usersGroups = ['A']`, then this returns early. OK

Am I right that case #3 should return early?

This might be more correct:
```suggestion
			if (count(array_intersect($usersGroups, $excludedGroups) !== 0) {
				return new DataResponse($this->result);
			}
```

Reply:
It's been a while since I've looked at Nextcloud, so please excuse any terminology mixups.

If I remember correctly my theory was that a user should be blocked from viewing potential Sharees only if all the groups they are a member of are blocked from sharing. (https://github.com/nextcloud/server/issues/20950#issuecomment-783339224)

I was looking at it from a scenario where some imaginary Admin user is also a member of a group which is blocked from sharing: should membership of a single excludedGroup prevent a user from getting a list of sharees? At the time Share Manager took the approach that a user should be blocked from sharing only if they were only a member of excluded groups so I followed that precedent (appears to still be the decision made in https://github.com/nextcloud/server/blob/master/lib/private/Share20/ShareDisableChecker.php#L52)

Either way, that's probably a higher level design decision that I'll leave to someone else.

Reply:
Yes sharing is only disabled if the user is **only** in groups disabled for sharing.

Reply:
I think it can be simplified to just:

```
if ($this->shareManager->sharingDisabledForUser($this->userId)) {
	return new DataResponse($this->result);
}
```